src/tracks/InstrumentTrack.cpp

```diff
@@ -653,7 +653,9 @@ bool InstrumentTrack::play( const MidiTime & _start, const fpp_t _frames,
 	{
 		Pattern* p = dynamic_cast<Pattern*>( *it );
 		// everything which is not a pattern or muted won't be played
-		if( p == NULL || ( *it )->isMuted() )
+		if(p == NULL ||
+			(Engine::getSong()->playMode() == Song::Mode_PlaySong
```

Comment:
It will make muted patterns in a Beat/Bassline pattern play sound when playing the BB pattern.
```suggestion
			(Engine::getSong()->playMode() != Song::Mode_PlayPattern
```
Also, could you add a comment describing this behavior?

Reply:
I think allowing a BB Editor pattern to be played as well is a good change, too, then. There is no indication that I can see that tells you the pattern is muted, so behavior of pressing play in their own window should be "I hear sound" rather than "I hear nothing". Ideally I believe the only time you shouldn't hear the pattern play is from the Song Editor, any other place you should allow it to play.